Assets/MixedRealityToolkit/_Core/Boundary/MixedRealityBoundarySystem.cs

```diff
@@ -66,35 +71,141 @@ public override void Reset()
             InitializeInternal();
         }
 
+        public bool Contains(Vector3 location)
+        {
+            return Contains(location, Boundary.Type.TrackedArea);
+        }
+
+        /// <inheritdoc/>
+        public bool Contains(Vector3 location, Boundary.Type boundaryType)
+        {
+            if (!EdgeUtilities.IsValidPoint(location))
+            {
+                // Invalid location.
+                return false;
+            }
+
+            if (!FloorHeight.HasValue)
+            {
+                // No floor.
+                return false;
+            }
+
+            // Handle the user teleporting (boundary moves with them).
+            location = CameraCache.Main.transform.parent.InverseTransformPoint(location);
+
+            if ((FloorHeight.Value > location.y) ||
+                (BoundaryHeight < location.y))
+            {
+                // Location below the floor or above the boundary height.
+                return false;
+            }
+
+            // Boundary coordinates are always "on the floor"
+            Vector2 point = new Vector2(location.x, location.z);
+
+            if (boundaryType == Boundary.Type.PlayArea)
+            {
+                // Check the inscribed rectangle.
+                if (rectangularBounds != null)
+                {
+                    return rectangularBounds.IsInsideBoundary(point);
+                }
+            }
+            else if(boundaryType == Boundary.Type.TrackedArea)
+            {
+                // Check the geometry
+                return EdgeUtilities.IsInsideBoundary(Bounds, point);
+            }
+
+            // Not in either boundary type.
+            return false;
+        }
+
+        /// <inheritdoc/>
+        public bool TryGetRectangularBoundsParams(
+            out Vector2 center, 
+            out float angle, 
```

Comment:
These could be a single line.

Reply:
i find it harder to read that way, but siure